Add stations support

lenisko opened the pull request:
It's working, but would be great to take a peak at queries (if I didn't miss anything).

lenisko committed:
feat: stations

lenisko committed:
feat: stations

lenisko committed:
fix(stations): fetch only those with endTime greater than now

TurtIeSocks reviewed (commented):
Seems mostly fine, just a couple of small comments.

lenisko commented on server/model/src/db/sea_orm_active_enums.rs:
```diff
@@ -19,6 +19,10 @@ pub enum Type {
+    #[sea_orm(string_value = "circle_station")]
+    CircleStation,
+    #[sea_orm(string_value = "circle_smart_station")]
+    CircleSmartStation,
```
No clue! Copy-pasted pokemon.

Fabio1988 replied:
RDM db doesn't even support station :)

lenisko committed:
refactor(stations): drop CircleSmartStation variant

Stations only use the non-smart CircleStation mode; remove the unused
smart variant from server enums, parsing/match arms, calculate route,
and client constants/utils.

lenisko committed:
refactor(stations): drop RDM support for stations

RDM scanner has no circle_station instance type; remove it from
RDM_ROUTES on the client and reject CircleStation in RDM instance
upsert. Unown paths (area, calculate, queries) untouched.

TurtIeSocks committed:
fix: claude's review
